akka-actor/src/main/scala/akka/dispatch/AbstractDispatcher.scala

```diff
@@ -377,16 +377,8 @@ object ForkJoinExecutorConfigurator {
                                threadFactory: ForkJoinPool.ForkJoinWorkerThreadFactory,
                                unhandledExceptionHandler: Thread.UncaughtExceptionHandler)
     extends ForkJoinPool(parallelism, threadFactory, unhandledExceptionHandler, true) with LoadMetrics {
-    override def execute(r: Runnable): Unit = {
-      if (r eq null) throw new NullPointerException("The Runnable must not be null")
-      val task =
-        if (r.isInstanceOf[ForkJoinTask[_]]) r.asInstanceOf[ForkJoinTask[Any]]
```

Comment:
Really tackling this means talking with Doug and possibly rolling our own variant of FJP, the current revert is only a stop gap measure to reduce risk. "talking with Doug" is not something that we track on github, though ;-)